joomla/joomla-cms
# [Admin Menu] Fix showing of protected menu items in container type menu item

### Summary of Changes
Removed bulk actions from protected menu items. Kept the list intact in readonly state, we may decide to remove it if that sounds better.

Updated `MenusHelper` and `JHtmlMenus` to respect `client_id` wherever applicable.

Menu item of type `container` now allows users to select items from the menutype `main` which they want to hide from their custom admin menu. Any new extension installed afterwards will be displayed until explicitly marked to be hidden.

The bug is fixed which caused container to show up (though empty) even when there were no child items to it. This can be useful to hide the container itself by hiding all its items. It will automatically appear back when a new component is installed.


### Testing Instructions
* Create a menutype 'menu', you should be able to create it without any warning after this patch. Without this patch a warning will be shown saying it is a reserved name.

* Menu manager menutype dropdown filter should no longer shown the options **Menu (protected)** and **Main (protected)**.

* ~Edit/Publish/Unpublish/Delete/Trash/Batch toolbar buttons should not be displayed in menu manager when filtered by the protected menutype 'main'.~

* Create a new custom admin menu and add a new menu item of type **System Links > Components Menu Container**. You should be able to choose menu items to **hide**.

* The above selected menu items should not show under this container menu item when using your custom admin menu.

* Hide everything in the container, the container should itself be hidden as well. 

* Install a new extension. The new extension should show up in the container. 

* The issues #13787 and #13788 were already fixed. Make sure this patch does not break it again.


### Documentation Changes Required
None

Pinging @infograf768
